databases/schemas/modlog.sql

```diff
@@ -0,0 +1,29 @@
+ CREATE TABLE IF NOT EXISTS modlog (
+    modlog_id INTEGER NOT NULL PRIMARY KEY,
+    -- UNIX timestamp
+    timestamp INTEGER NOT NULL,
+    --- roomid OR global-roomid
+    roomid TEXT NOT NULL,
+    action TEXT NOT NULL,
+    visual_roomid TEXT,
+    action_taker_userid TEXT,
+    userid TEXT,
+    autoconfirmed_userid TEXT,
+    ip TEXT,
+    note TEXT
+);
+
+CREATE TABLE IF NOT EXISTS alts (
+    modlog_id INTEGER NOT NULL,
+    userid TEXT NOT NULL,
+    PRIMARY KEY (modlog_id, userid),
+    FOREIGN KEY (modlog_id) REFERENCES modlog(modlog_id)
+);
+
+CREATE INDEX IF NOT EXISTS ml_index_1 ON modlog(timestamp);
+CREATE INDEX IF NOT EXISTS ml_index_2 ON modlog(roomid, userid, autoconfirmed_userid, timestamp);
```

Comment:
This index seems unnecessary and I don't think any query in the code would be able to use that (a query to make use of `autoconfirmed_userid` would have to match `userid` in the same row, for example `WHERE roomid = ? AND userid = ? AND autoconfirmed_userid = ?`). You have a query that does `userid = ? OR autoconfirmed_userid = ?`, but that's won't be able to use this index.